momepy/functional/tests/test_diversity.py

```diff
@@ -115,3 +116,141 @@ def test_describe_array(self):
         r2 = mm.describe(area.values, self.graph)
 
         assert_frame_equal(r, r2)
+
+
+class TestDescribeReached:
+    def setup_method(self):
+        test_file_path = mm.datasets.get_path("bubenec")
+        self.df_buildings = gpd.read_file(test_file_path, layer="buildings")
+        self.df_streets = gpd.read_file(test_file_path, layer="streets")
+        self.df_tessellation = gpd.read_file(test_file_path, layer="tessellation")
+        self.df_streets["nID"] = mm.unique_id(self.df_streets)
+        self.df_buildings["height"] = np.linspace(10.0, 30.0, 144)
+        self.df_tessellation["area"] = self.df_tessellation.geometry.area
+        self.df_buildings["area"] = self.df_buildings.geometry.area
+        self.df_buildings["fl_area"] = mm.FloorArea(self.df_buildings, "height").series
+        self.df_buildings["nID"] = mm.get_network_id(
+            self.df_buildings, self.df_streets, "nID"
+        )
+        blocks = mm.Blocks(
+            self.df_tessellation, self.df_streets, self.df_buildings, "bID", "uID"
+        )
+        self.blocks = blocks.blocks
+        self.df_buildings["bID"] = blocks.buildings_id
+        self.df_tessellation["bID"] = blocks.tessellation_id
+        self.graph = (
+            Graph.build_contiguity(self.df_streets.set_index("nID"), rook=False)
+            .higher_order(k=2, lower_order=True)
+            .assign_self_weight()
+        )
+
+    def test_describe_reached_input(self):
+        with pytest.raises(
+            ValueError,
+            match=("One of result_index or graph has to be specified, but not both."),
+        ):
+            mm.describe_reached(self.df_buildings[["area"]], self.df_buildings["nID"])
+
+        with pytest.raises(
+            ValueError,
+            match=("One of result_index or graph has to be specified, but not both."),
+        ):
+            mm.describe_reached(
+                self.df_buildings[["area"]],
+                self.df_buildings["nID"],
+                result_index=self.df_streets.index,
+                graph=self.graph,
+            )
+
+    def test_describe_reached(self):
+        df = mm.describe_reached(
+            self.df_buildings[["area", "fl_area"]],
+            self.df_buildings["nID"],
+            self.df_streets.index,
+        )
+        # not using assert_result since the method
+        # is returning an aggregation, indexed based on nID
```

Comment:
Why not? We use it in `describe` without issues. I'd prefer to use it as it checks the index.